### ISSUE TITLE
Build a dApp for Deploying Stable Liquidity Pools (15 ETH)

### ISSUE THREAD

**Issue by yudilevi** [bancorprotocol/contracts#310]
Title:
Build a dApp for Deploying Stable Liquidity Pools

Prize:
15 ETH

Challenge Description:
Bancor challenges developers to hack on one of the most exciting DeFi primitives - the decentralized liquidity pool. Liquidity pools are designed to perform non-custodial token-to-token conversions that also accrue value for liquidity providers who effectively act as passive market makers.

Up until now, liquidity pools have mostly been deployed with two volatile assets in their reserves. This has required liquidity providers to speculate on two volatile assets when contributing liquidity.

Bancor recently announced the release of USDB - a stable version of Bancor’s Network Token, BNT. USDB-based relays allow liquidity providers to plug into Bancor and generate fees from conversions without exposure to two volatile assets - which could dramatically reduce the risk of impermanent loss.

Now, we’re looking for a pioneering developer to create a DAPP that allows liquidity providers to transfer their liquidity from an existing Bancor liquidity pool (using BNT) to a Bancor liquidity pool that utilizes USDB.

Submission Requirements
For a certain token X, the dApp should enable the following process:

- Create a new XUSDB liquidity pool in addition to the existing XBNT relay (if such a relay doesn’t already exist)
- Allow the user to determine the fee on the new XUSDB liquidity pool
- Liquidate the user’s X Relay Tokens (the user should be able to enter an amount for partial liquidation)
- Convert the resulting BNT tokens into USDB
- Fund the new XUSDB liquidity pool
- Add controls that enable liquidity pool owners to activate/deactivate the BNT-based and USDB-based pools.

Important notes:
- Converting BNT to USDB should be done via Bancor Protocol, not opening a CPD.
- The newly deployed converters should be recorded in a registry (e.g., the ConverterRegistry contract) for on-chain discovery.

Judging Criteria:
- Best submission meeting all the criteria.
- Clean code & tech design.
- If applicable - clear UI, easy to use and generally good user experience.

Submission Deadline
October 4, 2019.

Winner Announcement Date
October 10, 2019.

Helpful tools:
Bancor Github: https://github.com/bancorprotocol/contracts
USDB Github: https://github.com/zachdoty/peg-usdb
Open-source liquidity portal built atop Bancor: bancor.cotrader.com & GitHub: https://github.com/CoTraderCore/Bancor-CoTrader-UI
Bancor Developer’s Telegram Channel: https://t.me/BancorDevelopers
Video explainer on Bancor community staking:
https://www.youtube.com/watch?v=oXTvHwRO61M&feature=youtu.be
Video explainer on Bancor liquidity pools (i.e., “Bancor Relays”):
https://blog.bancor.network/how-bancor-relays-work-c712a374374f
More resources at blog.bancor.network

**Comment by gitcoinbot** [comment]
Issue Status: 1. **Open** 2. Started 3. Submitted 4. Done 

<hr>

__This issue now has a funding of 15.0 ETH (2536.08 USD @ $169.07/ETH)  attached to it as part of the yudilevi fund.__

 * If you would like to work on this issue you can 'start work' [on the Gitcoin Issue Details page](https://gitcoin.co/issue/bancorprotocol/contracts/310/3502).
* Want to chip in? Add your own contribution [here](https://gitcoin.co/issue/bancorprotocol/contracts/310/3502).
* Questions? Checkout <a href='https://gitcoin.co/help'>Gitcoin Help</a> or the <a href='https://gitcoin.co/slack'>Gitcoin Slack</a>
* $150,624.46 more funded OSS Work available on the [Gitcoin Issue Explorer](https://gitcoin.co/explorer)

**Comment by gitcoinbot** [comment]
Issue Status: 1. **Open** 2. Started 3. Submitted 4. Done 

<hr>

__Work has been started__.


These users each claimed they can complete the work by 1 month from now.
Please review their action plans below:


**1) [mul1sh](https://gitcoin.co/profile/mul1sh) has started work.**

I've been going through the bancor liquidity docs and understand them quite well, so i can definitely build this dApp as required. Thanks

Learn more [on the Gitcoin Issue Details page](https://gitcoin.co/issue/bancorprotocol/contracts/310/3502).

**Comment by gitcoinbot** [comment]
Issue Status: 1. **Open** 2. Started 3. Submitted 4. Done 

<hr>

__Work for 15.0 ETH (2628.75 USD @ $175.25/ETH) has been submitted by__:

 1. [@mul1sh](https://gitcoin.co/profile/mul1sh)



@yudilevi please take a look at the submitted work:
* [PR](https://github.com/mul1sh/fluid) by @mul1sh

<hr>

* Learn more [on the Gitcoin Issue Details page](https://gitcoin.co/issue/bancorprotocol/contracts/310/3502)
* Want to chip in? Add your own contribution [here](https://gitcoin.co/issue/bancorprotocol/contracts/310/3502).
* Questions? Checkout <a href='https://gitcoin.co/help'>Gitcoin Help</a> or the <a href='https://gitcoin.co/slack'>Gitcoin Slack</a>
* $151,072.41 more funded OSS Work available on the [Gitcoin Issue Explorer](https://gitcoin.co/explorer)

**Comment by tgrecojs** [comment]
@gitcoinbot is this still taking submissions? gitcoin says to check to see if submissions are being added. I spent a good bit of time trying to get this in order but ran into a few issues as I was unfamiliar with the API.  Specifically, I don't have test funds in order to complete the transactions . I'm not sure if this is something i'm misunderstandings so I reached out to the Bancor team but I didn't have any luck as they were unable to supply me with one.

I've already built out a wallet that provides the ability to toggle between currencies, but i'm really just stuck with this testing. Fortmatic does supply 1 test ETH so simulate single and multi party trades. 

If there was ever a way in which I could set up a discussion with someone from Bancor, then it would be really appreciated as I feel like it would clear up a lot for me. I'm new to crytpo, but work in UI development of exceedingly high volume trading applications at a tranditional finalicial insittution. I'm really interested in this new landscape though! Appreciate Bancor for even holding the content - definitely let me know if I could still submit my work for a valid entry (if others could submit as well). Thanks :)
